frontend/packages/console-app/src/components/quick-starts/catalog/QuickStartCatalogFilter.tsx

```diff
@@ -0,0 +1,118 @@
+} from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import { useQueryParams } from '@console/shared';
+import { removeQueryArgument, setQueryArgument } from '@console/internal/components/utils';
+import { QuickStartStatus } from '../utils/quick-start-types';
+import { QUICKSTART_SEARCH_FILTER_KEY, QUICKSTART_STATUS_FILTER_KEY } from '../utils/const';
+
+import './QuickStartCatalogFilter.scss';
+
+type QuickStartCatalogFilterProps = {
+  quickStartsCount: number;
+  quickStartStatusCount: Record<QuickStartStatus, number>;
+};
+
+const QuickStartCatalogFilter: React.FC<QuickStartCatalogFilterProps> = ({
+  quickStartsCount,
+  quickStartStatusCount,
+}) => {
+  const { t } = useTranslation();
+  const queryParams = useQueryParams();
+  const searchQuery = queryParams.get(QUICKSTART_SEARCH_FILTER_KEY) || '';
+  const statusFilters = queryParams.get(QUICKSTART_STATUS_FILTER_KEY)?.split(',') || [];
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+
+  const onRowfilterSelect = React.useCallback(
+    (e) => {
+      setIsDropdownOpen(false);
+      const selectedFiltersList = statusFilters.includes(e.target.id)
+        ? statusFilters.filter((status) => status !== e.target.id)
+        : [...statusFilters, e.target.id];
+      if (selectedFiltersList.length > 0) {
+        setQueryArgument('status', selectedFiltersList.join(','));
+      } else {
+        removeQueryArgument(QUICKSTART_STATUS_FILTER_KEY);
+      }
+    },
+    [setIsDropdownOpen, statusFilters],
+  );
+
+  const handleTextChange = (val: string) => {
+    if (val.length > 0) {
+      setQueryArgument(QUICKSTART_SEARCH_FILTER_KEY, val);
+    } else {
+      removeQueryArgument(QUICKSTART_SEARCH_FILTER_KEY);
+    }
+  };
+
+  const statusTypes = {
+    [QuickStartStatus.COMPLETE]: t('quickstart~Complete'),
+    [QuickStartStatus.IN_PROGRESS]: t('quickstart~In progress'),
+    [QuickStartStatus.NOT_STARTED]: t('quickstart~Not started'),
+  };
+
+  const dropdownItems = Object.entries(statusTypes).map(([key, val]) => (
+    <DropdownItem id={key} key={key} className="co-filter-dropdown__item">
+      <div className="co-filter-dropdown-item">
+        <span className="co-filter-dropdown-item__checkbox">
+          <Checkbox isChecked={statusFilters.includes(key)} id={`${key}-checkbox`} />
+        </span>
+        <span className="co-filter-dropdown-item__name">{`${val} (${quickStartStatusCount[key]})`}</span>
+      </div>
+    </DropdownItem>
+  ));
+
+  const dropdownToggle = (
+    <DropdownToggle
+      onToggle={() => setIsDropdownOpen(!isDropdownOpen)}
+      toggleIndicator={CaretDownIcon}
+    >
+      {t('quickstart~Status')}
+      {statusFilters.length > 0 && (
+        <Badge className="co-quick-start-catalog-filter__status" isRead>
+          {statusFilters.length}
+        </Badge>
+      )}
+    </DropdownToggle>
+  );
+
+  return (
+    <Flex className="co-quick-start-catalog-filter__flex">
+      <FlexItem>
```

Comment:
You should set the width of this flex item and not the `SearchInput`. With the current implementation, the input doesn't shrink for mobile.
```
  flex-grow: 1;
  max-width: 500px;
```

Furthermore, can we use a patternfly `Toolbar` here?

Reply:
Updated.